code/game/antagonist/station/renegade.dm

```diff
@@ -49,9 +61,22 @@ var/datum/antagonist/renegade/renegades
 		return
 
 	var/gun_type = pick(spawn_guns)
-	var/obj/item/gun = new gun_type(get_turf(player))
-	player.put_in_hands(gun)
+	if(islist(gun_type))
+		gun_type = pick(gun_type)
 
+	var/obj/item/gun = new gun_type(get_turf(player))
+	var/list/slots = list (
+		"backpack" = slot_in_backpack,
+		"left hand" = slot_l_hand,
+		"right hand" = slot_r_hand,
+		)
+	for(var/slot in slots)
```

Comment:
I believe there is already a proc that will iterate over all slots and attempt to equip it. One of the hotkeys uses the same behavior. If it isn't a proc in that case it should probably be made one.